src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Builder/OpenTelemetryBuilderOtlpExporterExtensions.cs

```diff
@@ -0,0 +1,210 @@
+    {
+        Guard.ThrowIfNull(configure);
+
+        return UseOtlpExporter(builder, name: null, configure: configure);
+    }
+
+    internal static IOpenTelemetryBuilder UseOtlpExporter(
+        this IOpenTelemetryBuilder builder,
+        IConfiguration configuration)
+    {
+        Guard.ThrowIfNull(configuration);
+
+        return UseOtlpExporter(builder, name: null, configuration: configuration);
+    }
+
+    internal static IOpenTelemetryBuilder UseOtlpExporter(
+        this IOpenTelemetryBuilder builder,
+        string? name = null,
+        IConfiguration? configuration = null,
+        Action<OtlpExporterBuilder>? configure = null)
+    {
+        Guard.ThrowIfNull(builder);
+
+        if (configuration != null && string.IsNullOrEmpty(name))
+        {
+            name = "otlp";
+        }
+
+        var otlpBuilder = new OtlpExporterBuilder(builder.Services, name, configuration);
+
+        configure?.Invoke(otlpBuilder);
+
+        UseOtlpExporterInternal(builder, name);
+
+        return builder;
+    }
+
+    private static void UseOtlpExporterInternal(IOpenTelemetryBuilder builder, string? name)
+    {
+        // Note: We automatically turn on signals for "UseOtlpExporter"
+        builder
+            .WithLogging()
+            .WithMetrics()
+            .WithTracing();
+
+        var services = builder.Services;
+
+        // Note: UseOtlpExporterRegistration is added to the service collection
+        // to detect repeated calls to "UseOtlpExporter" and to throw if
+        // "AddOtlpExporter" extensions are called
+        services.AddSingleton<UseOtlpExporterRegistration>();
+
+        services.RegisterOptionsFactory(configuration => new SdkLimitOptions(configuration));
+        services.RegisterOptionsFactory(configuration => new ExperimentalOptions(configuration));
+        services.RegisterOptionsFactory((sp, configuration, name) => new OtlpExporterBuilderOptions(
+            configuration,
+            sp.GetRequiredService<IOptionsMonitor<SdkLimitOptions>>().CurrentValue,
+            sp.GetRequiredService<IOptionsMonitor<ExperimentalOptions>>().CurrentValue,
+            sp.GetService<IOptionsMonitor<LogRecordExportProcessorOptions>>()?.Get(name),
+            sp.GetService<IOptionsMonitor<MetricReaderOptions>>()?.Get(name),
+            sp.GetService<IOptionsMonitor<ActivityExportProcessorOptions>>()?.Get(name)));
+
+        name ??= Options.DefaultName;
+
+        services.ConfigureOpenTelemetryLoggerProvider(
+            (sp, logging) =>
+            {
+                var builderOptions = GetBuilderOptionsAndValidateRegistrations(sp, name);
+
+                var processor = OtlpLogExporterHelperExtensions.BuildOtlpLogExporter(
+                    sp,
+                    builderOptions.LoggingOptionsInstance.ApplyDefaults(builderOptions.DefaultOptionsInstance),
+                    builderOptions.LogRecordExportProcessorOptions ?? throw new NotSupportedException(),
```

Comment:
Could you add a comment here describing what it means to have `builderOptions.LogRecordExportProcessorOptions` as `null` here or why we throw a `NotSupportedException`?

Reply:
I added some notes: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5400/commits/b75f139910eef47fbf56644a0756045713bee107

It is really a left over thing from an earlier iteration where you could select the signals you wanted. But I think it is good to leave just in case we ever restore that?